sc/src/main/java/uk/ac/ebi/atlas/experimentpage/JsonExperimentTSnePlotController.java

```diff
@@ -76,14 +82,23 @@ public String tSnePlotWithMetadata(
             @RequestParam(defaultValue = "") String accessKey) {
         Experiment experiment = experimentTrader.getExperiment(experimentAccession, accessKey);
 
-        return GSON.toJson(
-                ImmutableMap.of(
-                        "series",
-                        modelForHighcharts(
-                                "",
-                                new TreeMap<>(
-                                        tSnePlotService.fetchTSnePlotWithMetadata(
-                                                experiment.getAccession(), perplexity, metadata)))));
+        // need to have this check in case there is no metadata for specified experiment
+        if(isNullOrEmpty(getTSnePlotMetadata(experiment.getAccession()))) {
```

Comment:
If the result of `getTsnePlotMetadata` is a `JsonArray`, why are you using `isNullOrEmpty` here, and `.size() == 0` later on? You have to be consistent in your implementation, especially in the same class when the discrepancy is so visible!

Reply:
The result of getTsnePlotMetadata is a String so that's why I have to treat return result as String to check if it is empty or null.

Regarding `.size()==0` here I'm dealing with JsonArray so depending on if it is empty or not I am returning null or actual value respectively.

Reply:
Sorry about that, I didn't notice it was a different `getTsnePlotMetadata` (the service has the same method name so I think I got confused). I think the logic can still be simplified, and this check might not even be needed (see my other comment).